react/finger-exercises/redux/02 - react-redux/src/app/screens/App/index.js

```diff
@@ -57,22 +40,44 @@ class App extends Component {
       <Fragment>
         <Navbar />
         <div className={styles.container}>
-          <Search onSearch={this.onSearch} />
-          {this.state.books.length ? (
-            this.state.books.map(this.renderBooks)
+          <Search />
+          {this.props.originalData.length ? (
+            this.props.books.map(this.renderBooks)
           ) : (
             <div className={styles.noData}>
               <h2 className={styles.title}>No Data</h2>
             </div>
           )}
         </div>
-        {this.state.bookSelected.length ? (
-          <ShoppingCart data={this.state.bookSelected} addItem={this.addItem} removeItem={this.removeItem} />
-        ) : null}
+        {this.props.bookSelected.length ? <ShoppingCart /> : null}
```

Comment:
Extract the keys of props and then you don't need to repeat this.props all the time